cql3/expr/expression.cc

```diff
@@ -1723,10 +1723,60 @@ std::optional<bool> get_bool_value(const constant& constant_val) {
     return constant_val.view().deserialize<bool>(*boolean_type);
 }
 
+cql3::raw_value evaluate(const binary_operator& binop, const evaluation_inputs& inputs) {
+    if (binop.order == comparison_order::clustering) {
+        throw exceptions::invalid_request_exception("Can't evaluate a binary operator with SCYLLA_CLUSTERING_BOUND");
+    }
+
+    bool binop_result;
+
+    switch (binop.op) {
+        case oper_t::EQ: {
+            binop_result = equal(binop.lhs, binop.rhs, inputs);
```

Comment:
This is not quite correct. equal() will return true/false, but SQL = can return true, false, or NULL.

So we need to change equal() (and like(), and similar) to return std::optional<bool> with a disengaged optional corresponding to NULL.